### PR TITLE
Add FLOAT_ZERO_HEX database function and tests

### PR DESCRIPTION
**Pull request by findolor**
fix https://github.com/rainlanguage/sqlite-web/issues/16

## Summary
- add a new `FLOAT_ZERO_HEX` SQLite scalar function that surfaces the canonical zero rain-math float hex string
- register the function alongside other custom database functions and add a wasm test to ensure the default hex matches parsing zero
- create Vitest coverage that exercises the new function end-to-end and update database cleanup helpers for the new table

## Testing
- cargo check -p sqlite-web-core --target wasm32-unknown-unknown *(fails: local `rain-math-float` source path missing)*
- npm install *(fails: repository requires Node.js >= 22 for @rainlanguage/float)*

------
https://chatgpt.com/codex/tasks/task_e_68cbb532e16883338defbbf23f7c753b

## Summary by CodeRabbit

- New Features
  - Added FLOAT_ZERO_HEX() (arity 0) SQLite function returning the canonical zero value as a hex string; usable in queries, storage, aggregation, and as a column default.

- Tests
  - Added tests for registration, repeated calls, canonical-zero retrieval, storage/aggregation with FLOAT_SUM, hex↔decimal conversions, and default-value usage.

- Chores
  - Test DB cleanup now drops float_zero_usage and float_zero_defaults tables.
  - Updated test dependency to a new package snapshot.

### CODE REVIEW

File: svelte-test/tests/database-functions/float-zero-hex.test.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    const sumResult = await db.query(
+      "SELECT FLOAT_SUM(amount) as total FROM float_zero_usage",
+    );
+    const sumData = JSON.parse(sumResult.value || "[]");
+    expect(sumData).toHaveLength(1);
+    expect(sumData[0].total).toBe(canonicalZeroHex);
+    expect(decodeFloatHex(sumData[0].total)).toBe("0");
+  });
```
[the remark this answers is not in the document]
Comment (findolor) [reply svelte-test/tests/database-functions/float-zero-hex.test.ts:85]: this is not needed. we don't have any arguments so it's a noop

File: packages/sqlite-web-core/src/database_functions/float_zero_hex.rs
Comment (0xgleb) [inline packages/sqlite-web-core/src/database_functions/float_zero_hex.rs:1]: should we add some tests to this module?

File: svelte-test/tests/database-functions/float-zero-hex.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xgleb) [reply svelte-test/tests/database-functions/float-zero-hex.test.ts:85]: It would be good to have error coverage since we don't get this from the compiler

File: svelte-test/tests/database-functions/float-zero-hex.test.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  it("should be registered and return the canonical zero hex string", async () => {
+    const result = await db.query("SELECT FLOAT_ZERO_HEX() as zero_hex");
+    const data = JSON.parse(result.value || "[]");
+    expect(data).toHaveLength(1);
+    expect(data[0].zero_hex).toBe(canonicalZeroHex);
+
+    const decimal = decodeFloatHex(data[0].zero_hex);
+    expect(decimal).toBe("0");
+  });
```
[the remark this answers is not in the document]
Comment (findolor) [reply svelte-test/tests/database-functions/float-zero-hex.test.ts:32]: no need to do this

File: svelte-test/tests/database-functions/float-zero-hex.test.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+  it("should be registered and return the canonical zero hex string", async () => {
+    const result = await db.query("SELECT FLOAT_ZERO_HEX() as zero_hex");
+    const data = JSON.parse(result.value || "[]");
+    expect(data).toHaveLength(1);
+    expect(data[0].zero_hex).toBe(canonicalZeroHex);
+
+    const decimal = decodeFloatHex(data[0].zero_hex);
+    expect(decimal).toBe("0");
```
[the remark this answers is not in the document]
Comment (findolor) [reply svelte-test/tests/database-functions/float-zero-hex.test.ts:31]: not needed

**Review by hardyjosh — APPROVED** [review]

**Review by 0xgleb — APPROVED** [review]
